server/src/repositories/session.repository.ts

```diff
@@ -1,56 +1,64 @@
+import { Insertable, Kysely, Updateable } from 'kysely';
+import { InjectKysely } from 'nestjs-kysely';
+import { DB, Sessions } from 'src/db';
 import { DummyValue, GenerateSql } from 'src/decorators';
-import { SessionEntity } from 'src/entities/session.entity';
+import { SessionEntity, withUser } from 'src/entities/session.entity';
 import { ISessionRepository, SessionSearchOptions } from 'src/interfaces/session.interface';
-import { LessThanOrEqual, Repository } from 'typeorm';
+import { asUuid } from 'src/utils/database';
+import { Repository } from 'typeorm';
 
 @Injectable()
 export class SessionRepository implements ISessionRepository {
-  constructor(@InjectRepository(SessionEntity) private repository: Repository<SessionEntity>) {}
+  constructor(
+    @InjectRepository(SessionEntity) private repository: Repository<SessionEntity>,
+    @InjectKysely() private db: Kysely<DB>,
+  ) {}
 
-  @GenerateSql({ params: [DummyValue.DATE] })
+  @GenerateSql({ params: [{ updatedBefore: DummyValue.DATE }] })
   search(options: SessionSearchOptions): Promise<SessionEntity[]> {
-    return this.repository.find({ where: { updatedAt: LessThanOrEqual(options.updatedBefore) } });
+    return this.db
+      .selectFrom('sessions')
+      .selectAll()
+      .where('sessions.updatedAt', '<=', options.updatedBefore)
+      .execute() as Promise<SessionEntity[]>;
   }
 
   @GenerateSql({ params: [DummyValue.STRING] })
   getByToken(token: string): Promise<SessionEntity | null> {
-    return this.repository.findOne({
-      where: { token },
-      relations: {
-        user: {
-          metadata: true,
-        },
-      },
-    });
+    return this.db
+      .selectFrom('sessions')
+      .innerJoinLateral(withUser, (join) => join.onTrue())
+      .selectAll('sessions')
+      .select((eb) => eb.fn.toJson('user').as('user'))
+      .where('sessions.token', '=', token)
+      .executeTakeFirst() as unknown as Promise<SessionEntity | null>;
   }
 
+  @GenerateSql({ params: [DummyValue.UUID] })
   getByUserId(userId: string): Promise<SessionEntity[]> {
-    return this.repository.find({
-      where: {
-        userId,
-      },
-      relations: {
-        user: true,
-      },
-      order: {
-        updatedAt: 'desc',
-        createdAt: 'desc',
-      },
-    });
+    return this.db
+      .selectFrom('sessions')
+      .innerJoinLateral(withUser, (join) => join.onTrue())
+      .selectAll('sessions')
+      .select((eb) => eb.fn.toJson('user').as('user'))
+      .where('sessions.userId', '=', userId)
+      .orderBy('sessions.updatedAt', 'desc')
+      .orderBy('sessions.createdAt', 'desc')
+      .execute() as any as Promise<SessionEntity[]>;
   }
 
-  create<T extends Partial<SessionEntity>>(dto: T): Promise<T & { id: string }> {
-    return this.repository.save(dto);
+  create(dto: Insertable<Sessions>): Promise<SessionEntity> {
+    return this.db.insertInto('sessions').values(dto).returningAll().executeTakeFirst() as Promise<SessionEntity>;
```

Comment:
You cannot cast the `undefined` away here. If you can guarantee that will always be defined (which is probably the case here), you can `executeTakeFirstOrThrow()`